### PR TITLE
Change marker generation

### PR DESCRIPTION
**Pull request by timvanoostrom**
During the process of implementing the MarkerClusterGroup component into a Map component, I came across a few things I'd like to improve. This is mainly about the place where the `L.Marker`s are constructed. I've separated the marker construction from the component so customized markers can be used.

This is a breaking change because the layer data has to be created outside of the component. The component will not just generate the layer from a bunch of `LatLngTuple[]`

Also I've added the options to pass events to add to the clusterLayer and a setInstance method for even more flexibility.

### CODE REVIEW

File: packages/arm-cluster/src/components/MarkerClusterGroup.tsx [hunk 1]
```diff
@@ -76,21 +100,37 @@ const MarkerClusterGroup: React.FC<Props> = ({
     return null
   }, [mapInstance, optionsOverrides])
 
+  useEffect(() => {
+    if (setInstance && markerClusterGroup) {
```
Comment (chrisvanmook) [inline packages/arm-cluster/src/components/MarkerClusterGroup.tsx:104]: You can replace it with this:
```suggestion
    if (!setInstance || !markerClusterGroup) {
      return undefined
    }
```
So then you won't need to check this in the cleanup fn

File: packages/arm-cluster/src/components/MarkerClusterGroup.tsx [hunk 2]
```diff
@@ -76,21 +100,37 @@ const MarkerClusterGroup: React.FC<Props> = ({
...
+    }
+    return () => {
+      if (setInstance && markerClusterGroup) {
+        setInstance(null)
+      }
+    }
+  }, [setInstance, markerClusterGroup])
+
+  useEffect(() => {
+    if (markerClusterGroup && events) {
```
Comment (chrisvanmook) [inline packages/arm-cluster/src/components/MarkerClusterGroup.tsx:115]: Same here

File: stories/src/examples/MarkerFilters.tsx [hunk 3]
```diff
@@ -22,41 +17,84 @@ const Panel = styled.div`
...
+        iconAnchor: L.point(10, 10),
+      }),
+    })
+    return markerObject
+  })
+}
+
+interface MarkerFiltersProps {
+  markersSource: L.LatLngTuple[]
+  clusterComponent: (markers: any) => ReactNode
```
Comment (chrisvanmook) [inline stories/src/examples/MarkerFilters.tsx:51]: can be of type `L.LatLngTuple[]` I guess, or maybe even `L.LatLng[]`

File: packages/arm-cluster/src/components/MarkerClusterGroup.tsx (hunk 1 above)
Comment (timvanoostrom) [reply packages/arm-cluster/src/components/MarkerClusterGroup.tsx:104]: I think it's easier to reason about variables that do exist, it reads a bit better.

File: stories/src/examples/MarkerFilters.tsx (hunk 3 above)
Comment (timvanoostrom) [reply stories/src/examples/MarkerFilters.tsx]: It's actually `L.Marker[]`

File: packages/arm-cluster/src/components/MarkerClusterGroup.tsx (hunk 1 above)
Comment (chrisvanmook) [reply packages/arm-cluster/src/components/MarkerClusterGroup.tsx:104]: It's a matter of personal preference, indeed. But if you use the statement like I suggested, you won't need to duplicate the statement in the cleanup function :) Therefore more clean and still readable

Comment (timvanoostrom) [reply packages/arm-cluster/src/components/MarkerClusterGroup.tsx]: Good argument. I've changed all the effects in this file to this approach.

**Review by chrisvanmook — APPROVED** [review]

**Pull request merged**